### PR TITLE
Report on failures in logs

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Checking that the number of published articles is above a certain threshold does not give enough information to diagnose the error. In this PR, we had a check before that one that will look into the logs for the last two runs and filter through the ones about failures. If there's any, an email will be sent with the exact content of those logs.

TODO
- [x] Make sure log streams belong to the day's logs
- [x] Loop over the log stream array

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
[Kindle Incident retro actions](https://trello.com/c/73ImgbZg/670-kindle-incident-retro-actions)

File: src/checkPublication.ts [hunk 1]
```diff
@@ -1,93 +1,171 @@
...
+                    return;
+                } else if (data.events.length < 20) {
+                    throw new Error(`Log stream ${logStream.logStreamName} in ${logGroupName} only contains ${data.events.length} but we expected hundreds of them`)
+                }
+
+                resolve(data.events);
+            })
+        });
+        return getLogsFor(logStreams[0]).catch(() => getLogsFor(logStreams[1]));
+    }
```
Comment (JustinPinner) [inline src/checkPublication.ts]: `;`?

File: src/checkPublication.ts [hunk 2]
```diff
@@ -1,93 +1,171 @@
...
+
+    const checkLogs = (allLogs: Array<AWS.CloudWatchLogs.OutputLogEvent>): Promise<void> => {
+        const errors = allLogs.filter(log => / WARN | ERROR | FATAL /.test(log.message))
+        if (errors.length > 0) {
+            return Promise.reject(errors)
+        } else {
+            return Promise.resolve();
+        }
+    }
```
Comment (JustinPinner) [inline src/checkPublication.ts]: another one (`;`) - probably not essential these days but consistency FTW! And on line 144 I think 🙄

**Comment by JustinPinner** [comment]
Just to help me understand: will we still output the edition if we find warnings or errors in the logs, or can't find the log files? Does this just relate to the emails that are sent?

The reason for asking is that I wonder if the logs are guaranteed to be available and complete by the time we try to inspect them. IIRC this used to be unreliable, which is why the article count became the primary quality check.

**Comment by regiskuckaertz** [comment]
Sorry about the formatting mess 😞 

> will we still output the edition if we find warnings or errors in the logs, or can't find the log files? Does this just relate to the emails that are sent?

Just the emails.

> The reason for asking is that I wonder if the logs are guaranteed to be available and complete by the time we try to inspect them.

I didn't think about that. Perhaps the lambda trigger should be different; I'll have a look.

**Comment by regiskuckaertz** [comment]
Tested this locally best I could: it's working. Just made sure no email was being sent when doing so 😆

**Review by JustinPinner — APPROVED** [review]
Ok, based on an off-line conversation with Regis, in which I learned that this doesn't impair the actual edition process at all (just post-production checking and reporting) I think it's good to go 👍

**Pull request merged**